### PR TITLE
Issue 11 initial check in (#1321)

### PR DESCRIPTION
**Pull request by thefazzer**
**Issue 11 @ CHT Tests #1321**       
When checkpoints are enabled and the first is at block X,
when we present headers before that- none marked for download.
When we first present checkpointed header and some after only
headers before the first checkpoint(including it) are marked for download.
@Aprogiena @noescape00  pls review

### CODE REVIEW

**Comment by noescape00** [comment]
Conflicts

**Comment by thefazzer** [comment]
@noescape00 merged - I should be a few commits ahead now

**Comment by Aprogiena** [comment]
There is no code in this PR

**Comment by thefazzer** [comment]
@Aprogiena - resolved now, Issue 11 is present.

File: src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs [hunk 1]
```diff
@@ -676,6 +676,74 @@ public void PresentChain_CheckpointsEnabled_MarkToDownloadWhenCheckpointPresente
...
+            connectNewHeadersResult.DownloadTo.HashBlock.Should().Be(checkpoint1.Header.GetHash());
+
+            // Checking from first checkpoint back to the initialized chain (h20 -> h6).
+            ChainedHeader chainedHeader = chainedHeaderTree.GetChainedHeadersByHash()
+                .SingleOrDefault(x => (x.Value.HashBlock == checkpoint1.Header.GetHash())).Value;
+            while (chainedHeader.Height > initialChainSize)
+            {
+                chainedHeader.BlockValidationState.Should().Be(ValidationState.AssumedValid);
+                chainedHeader = chainedHeader.Previous;
+            }
```
Comment (Aprogiena) [inline src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs]: All looks good, but again we would like to check that headers above the first checkpoint still maintain the original validation state.

Comment (thefazzer) [reply src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs]: @Aprogiena - Done

**Comment by Aprogiena** [comment]
Looks good to me, @noescape00 your turn

**Pull request merged**